src/cuda/device.c

```
@@ -21,7 +21,11 @@ CUresult cuDeviceGet(CUdevice *device,int ordinal){
 CUresult cuDeviceGetCount( int* count ) {
     LOG_DEBUG("into cuDeviceGetCount");
     CUresult res = CUDA_OVERRIDE_CALL(cuda_library_entry,cuDeviceGetCount,count);
-    LOG_DEBUG("cuDeviceGetCount res=%d count=%d",res,*count);
+    if (res == CUDA_SUCCESS) {
+        LOG_DEBUG("cuDeviceGetCount res=%d count=%d",res,*count);
```

Cppcheck (2.21.0)

[warning] If resource allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfResources)